sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AccessTokenManager.java

```diff
@@ -0,0 +1,256 @@
+            return accessToken;
+        } else {
+            // TODO: incremental consent.
+            String idToken = getIdTokenFromSecurityContext();
+            return getAccessToken(idToken, applicationIdUri, uniformedPermissions);
+        }
+    }
+
+    private String getIdTokenFromSecurityContext() {
+        return Optional.of(SecurityContextHolder.getContext())
+                       .map(SecurityContext::getAuthentication)
+                       .map(Authentication::getPrincipal)
+                       .map(p -> (OidcUser) p)
+                       .map(OidcUser::getIdToken)
+                       .map(AbstractOAuth2Token::getTokenValue)
+                       .orElse(null);
+    }
+
+    /**
+     * Get AccessToken token for Microsoft Graph with expected permissions.
+     *
+     * @param idToken The token used to perform an get token request.
+     * @param permissions The permissions of resources to be authorized with, need to be formatted as lowercase.
+     * @return The access token for Microsoft Graph service.
+     * @throws ServiceUnavailableException If fail to acquire the token.
+     * @throws MsalServiceException If {@link MsalServiceException} has occurred.
+     */
+    public AccessToken getAccessTokenForMicrosoftGraph(String idToken,
+                                                       Set<String> permissions) throws ServiceUnavailableException {
+        return getAccessToken(idToken, aadAuthenticationProperties.getGraphApiUri(), permissions);
+    }
+
+    /**
+     * Get AccessToken for a resource server.
+     *
+     * @param idToken The token used to perform an get token request.
+     * @param applicationIdUri The Application ID URI of resource server, e.g., https://graph.microsoft.com for
+     * Microsoft Graph API.
+     * @param permissions The permissions of resources to be authorized with, need to be formatted as lowercase.
+     * @return The access token for resource server.
+     * @throws ServiceUnavailableException If fail to acquire the token.
+     * @throws MsalServiceException If {@link MsalServiceException} has occurred.
+     */
+    public AccessToken getAccessToken(String idToken,
+                                      String applicationIdUri,
+                                      Set<String> permissions) throws ServiceUnavailableException {
+        IAuthenticationResult result = getIAuthenticationResult(idToken, applicationIdUri, permissions);
+        Set<String> uniformedPermissions =
+            Arrays.stream(result.scopes().toLowerCase(Locale.ENGLISH).split(" "))
+                  .map(s -> s.startsWith(applicationIdUri) ? s.split(applicationIdUri)[1] : s)
+                  .collect(Collectors.toSet());
+        AccessToken accessToken = new AccessToken(
+            applicationIdUri,
+            uniformedPermissions,
+            result.expiresOnDate(),
+            result.accessToken()
+        );
+        accessToken.getAccessTokenWithRefreshAutomatically();
```

Comment:
We should call saveAccessTokenToSession here instead of putting it into getAccessTokenWithRefreshAutomatically(), because in that way tokens won't be saved until they are expired.